lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -145,22 +183,52 @@ export default function SubmitGradeForm({
     }
   };
 
-  const handleInput = useCallback(
-    (e: Event) => {
-      // If any input is detected, close the ValidationMessage.
-      setValidationError(false);
-      setGradeSaved(false);
+  const handleInputCommon = useCallback((e: Event): string => {
+    // If any input is detected, close the ValidationMessage.
+    setValidationError(false);
+    setGradeSaved(false);
 
-      const newDraftGradeValue = (e.target as HTMLInputElement).value;
-      setDraftGradeValue(newDraftGradeValue);
+    return (e.target as HTMLInputElement).value;
+  }, []);
+  const handleGradeInput = useCallback(
+    (e: Event) => {
+      const newGrade = handleInputCommon(e);
+      setDraftGradeValue(newGrade);
 
       // Check if there are unsavedChanges
-      onUnsavedChanges?.(newDraftGradeValue !== grade.data);
+      onUnsavedChanges?.(
+        newGrade !== grade.data?.grade ||
+          (draftCommentValue !== null &&
+            draftCommentValue !== grade.data.comment)
+      );
     },
-    [grade.data, onUnsavedChanges]
+    [
+      handleInputCommon,
+      onUnsavedChanges,
+      grade.data?.grade,
+      grade.data?.comment,
+      draftCommentValue,
+    ]
   );
+  const handleCommentInput = useCallback(
+    (e: Event) => {
+      const newComment = handleInputCommon(e);
+      setDraftCommentValue(newComment);
 
-  const disabled = !student;
+      // Check if there are unsavedChanges
+      onUnsavedChanges?.(
+        newComment !== grade.data?.comment ||
+          (draftGradeValue !== null && draftGradeValue !== grade.data.grade)
+      );
+    },
+    [
+      handleInputCommon,
+      onUnsavedChanges,
+      grade.data?.comment,
+      grade.data?.grade,
+      draftGradeValue,
+    ]
+  );
```

Comment:
I don't love these three event handlers (`handleInputCommon`, `handleGradeInput` and `handleCommentInput`).

I tried to have a single handler that could cover any input, but it ends up being a bit convoluted, as you have to parameterize the setter to be called, but also how the argument for `onUnsavedChanges` is computed.

I'm also thinking maybe the draft grade and the draft comment should be tracked differently, via a reducer instead of separated state pieces.